src/main/asciidoc/_chapters/ops_mgt.adoc

```diff
@@ -3479,6 +3478,27 @@ To enable ACL, add the following to your hbase-site.xml and restart your Master:
 <property>
 ----
 
+=== Migrating From Old Implementation
+The coprocessor `org.apache.hadoop.hbase.rsgroup.RSGroupAdminEndpoint` is
+deprected, but for compatible, if you want the pre 3.0.0 hbase client/shell
+to communicate with the new hbase cluster, you still need to add this
+coprocessor to master. And if this coprocessor is specified, the
+`hbase.balancer.rsgroup.enabled` flag will be set to true automatically to
+enable rs group feature.
+
+The main difference comparing to the old implementation is that, now the
+rsgroup for a table is stored in TableDescriptor, instead of in RSGroupInfo,
+so the getTables method of RSGroupInfo has been deprecated. And if you use the
+Admin methods to get the RSGroupInfo, its getTables method will always return
+empty. Of course the behavior for the old RSGroupAdminEndpoint is not changed,
+we will fill the tables field of the RSGroupInfo before returning, to make it
+compatible with old hbase client/shell.
+
+When upgrading, the migration between the RSGroupInfo and TableDescriptor will
```

Comment:
what happens wrt group enforcement during the migration? Will regions for grouped tables/namespaces go outside their assigned servers? Will rsgroup administrative commands I run wait to handle moving things in or out of a rsgroup until the migration is done?

Reply:
Just update the table descriptor, no other effect, you can see the comment in RSGroupInfoManagerImpl.migrate method for more details why this could work.

Reply:
excellent. please add a sentence here that expressly calls out that enforcement will keep working during the migration and running new admin commands is fine.

probably I will get folks asking me how to tell when the migration is complete, so if there's an easy answer to that we should include it.